Bottom: a70e486d7fb3d18adace33aea325353d847f6b41
Top:    799d9bfcd5b6545c40e5e19c7b50907bd25ed616
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-07 11:26:32 -0700

Refresh of fstat

---

diff --git a/src/preload.c b/src/preload.c
index b9c32e6..749857e 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -970,7 +970,7 @@ int fstat(int socket, struct stat *buf)
 	if (fd_get(socket, &fd) == fd_rsocket) {
 		ret = real.fstat(socket, buf);
 		if (!ret)
-			buf->st_mode |= __S_IFSOCK;
+			buf->st_mode = (buf->st_mode & ~S_IFMT) | __S_IFSOCK;
 	} else {
 		ret = real.fstat(fd, buf);
 	}
